snouter/src/auth/role.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable, ExecutionContext, CanActivate } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+
+@Injectable()
+export class RoleGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  matchRoles(roles: string[], userRole: string) {
+    return roles.some((role) => role === userRole);
+  }
```

Comment:
for this role detection you only match user role with one of route roles and that means if you want admins to be able to use app you need to either add Roles.ADMIN to every route (that defines roles) or extend this logic so admin can access their role routes and all under. second option means that route role would be minimum role level to access and adding new role later like superadmin would be trivial